clients/src/main/java/org/apache/kafka/common/requests/ProduceRequest.java

```diff
@@ -40,6 +40,7 @@
 import static org.apache.kafka.common.requests.ProduceResponse.INVALID_OFFSET;
 
 public class ProduceRequest extends AbstractRequest {
+    private static final short TRANSACTION_V2_MINIMAL_VERSION = 12;
```

Comment:
nit: should we have the last version before v2 to avoid the -1s in the methods? For isTransactionalV2Requested, we can have strictly greater than the version.